components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/template/APITemplateBuilderImpl.java

```diff
@@ -109,14 +152,57 @@ public void addHandler(String handlerName, Map<String, String> properties) {
 		addHandlerPriority(handlerName, properties, handlers.size());
 	}
 
-	public void addHandlerPriority(String handlerName, Map<String, String> properties, int priority) {
+	public void addHandlerPriority(String handlerName, Map<String, String> properties,
+								   int priority) {
 		HandlerConfig handler = new HandlerConfig(handlerName, properties);
 		handlers.add(priority, handler);
 	}
 
-	public String getTemplatePath() {
+	private String getVersionedWebAppTemplatePath() {
+		return "repository" + File.separator + "resources" + File.separator + "api_templates" +
+				File.separator +
+				APITemplateBuilderImpl.VELOCITY_TEMPLATE_SYNAPSE_CONFIG_VERSIONED_WEBAPP;
+	}
+
+	private String getNonVersionedWebAppTemplatePath() {
 		return "repository" + File.separator + "resources" + File.separator + "api_templates" +
-		       File.separator + APITemplateBuilderImpl.TEMPLATE_TYPE_VELOCITY + ".xml";
+				File.separator +
+				APITemplateBuilderImpl.VELOCITY_TEMPLATE_SYNAPSE_CONFIG_NON_VERSIONED_WEBAPP;
 	}
 
+	private String getVelocityLoggerName() {
+		AppManagerConfigurationService config =
+				ServiceReferenceHolder.getInstance().getAPIManagerConfigurationService();
+		String velocityLogPath = config.getAPIManagerConfiguration().getFirstProperty(
+				AppMConstants.VELOCITY_LOGGER);
+		if (velocityLogPath != null && velocityLogPath.length() > 1) {
+			return velocityLogPath;
+		} else {
+			return null;
+		}
+	}
+
+	private String processTemplate(VelocityEngine ve, VelocityContext vc, String templatePath)
+			throws APITemplateException {
+		StringWriter writer = new StringWriter();
+		try {
+			Template t = ve.getTemplate(templatePath);
+			t.merge(vc, writer);
+		} catch (ResourceNotFoundException e) {
+			String msg = "Cannot find Velocity template " + templatePath;
+			log.error(msg, e);
+			throw new APITemplateException(msg, e);
+		} catch (ParseErrorException e) {
+			String msg = "Cannot parse Velocity template " + templatePath;
+			log.error(msg, e);
+			throw new APITemplateException(msg, e);
+		} catch (IOException e) {
+			log.error("Cannot write processed Velocity template", e);
+			throw new APITemplateException("Cannot write processed Velocity template", e);
+		} catch (Exception e) {
```

Comment:
Instead of catching generic exceptions, a better option would be to use multi-catch statements and concretely declare all exceptions that are specifically caught within the context of this particular logic.